Remove version requirement in git-webkit for bugzilla and github

Constellation opened the pull request:
#### 1bbc66231a675bdb963ee6c1d7e1179869c8eda8
<pre>
Remove version requirement in git-webkit for bugzilla and github
<a href="https://bugs.webkit.org/show_bug.cgi?id=248163">https://bugs.webkit.org/show_bug.cgi?id=248163</a>
rdar://102571648

Reviewed by Darin Adler.

This patch removes asking version when creating a bugzilla entry from git-webkit since this was not happening in webkit-patch.
Since we do not have consensus on this addition, we should do the same thing to webkit-patch to reduce engineering burden.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
(Tracker):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py:
(TestBugzilla.test_create_prompt):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/github_unittest.py:
(TestGitHub.test_create_projects):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/branch_unittest.py:
(TestBranch.test_create_bug):

Canonical link: <a href="https://commits.webkit.org/256961@main">https://commits.webkit.org/256961@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8208017b4162f203ac3c8febaea5b8091ea3894d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/97365 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/6628 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/30526 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/106883 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/167147 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/101333 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/6935 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/35366 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/89774 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/103560 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/19/builds/103029 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/5208 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/83995 "Built successfully") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/32212 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/100941 "Found 1 webkitpy python3 test failure: webkitcorepy.tests.file_lock_unittest.FileLockTestCase.test_locked_timeout") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/87067 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/88882 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/75137 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/643 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/20348 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/20/builds/96301 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/625 "Built successfully") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/21812 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/5429 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/60/builds/44295 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/2360 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/1871 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/41157 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
```diff
@@ -434,13 +434,12 @@ def create(
-            )
+        if not version:
+            # This is the default option, aligned to webkit-patch behavior.
+            version = "WebKit Nightly Build"
```
I think this should be taken as an argument to the Bugzilla class and then our repositories bug definition in `metadata/trackers.json`...everything in `Tools/Scripts/libraries/webkitbugspy/webkitbugspy` has managed to be project agnostic so far, seems like a shame to break that now (even though bugzilla, in particular, is really only relevant to WebKit...)

Constellation replied:
OK, for now, I'll land it as is to fix the regression from webkit-patch, requiring the version in git-webkit, and adding a FIXME here.

darinadler reviewed (approved):
I think this is OK as is and possibly even better if you make the revision @JonWBedard requested. We should remove these labels entirely from GitHub too. I will do that after making sure nobody objects.

Constellation committed:
Remove version requirement in git-webkit for bugzilla and github
https://bugs.webkit.org/show_bug.cgi?id=248163
rdar://102571648

Reviewed by Darin Adler.

This patch removes asking version when creating a bugzilla entry from git-webkit since this was not happening in webkit-patch.
Since we do not have consensus on this addition, we should do the same thing to webkit-patch to reduce engineering burden.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
(Tracker):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py:
(TestBugzilla.test_create_prompt):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/github_unittest.py:
(TestGitHub.test_create_projects):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/branch_unittest.py:
(TestBranch.test_create_bug):

Canonical link: https://commits.webkit.org/256961@main
